### PR TITLE
"chef-client" => #{Chef::Dist::CLIENT}

### PR DESCRIPTION
**Pull request by bobchaos**
Signed-off-by: Marc Chamberland <mchamberland@pbsc.com>

### Description
More Changes related to https://github.com/chef/chef/issues/8376, this PR should clean up the word `chef-client` from the code base (perhaps a bit too aggressively).

### Issues Resolved
More work towards https://github.com/chef/chef/issues/8376

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] RELEASE\_NOTES.md has been updated if required (not required for bugfixes, required for API changes)
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

File: lib/chef/application.rb [hunk 1]
```diff
@@ -313,7 +314,7 @@ def fork_chef_client
             " finishing converge to exit normally (send SIGINT to terminate immediately)")
         end
 
-        client_solo = chef_config[:solo] ? "chef-solo" : "chef-client"
```
Comment (tas50) [inline lib/chef/application.rb]: we probably need a chef-solo variable as well

Comment (bobchaos) [reply lib/chef/application.rb]: I'll start a list to the proposal doc so we don't forget any.

Would you like for chef-solo be addressed in this same PR?

Comment (tas50) [reply lib/chef/application.rb]: It's not a blocker for this. It should just get done at some point

File: lib/chef/knife/bootstrap.rb [hunk 2]
```diff
@@ -417,7 +418,7 @@ def initialize(argv = [])
       # @return [String] Default bootstrap template
       def default_bootstrap_template
         if target_host.base_os == :windows
-          "windows-chef-client-msi"
+          "windows-#{Chef::Dist::CLIENT}-msi"
```
Comment (bobchaos) [inline lib/chef/knife/bootstrap.rb]: I wasn't sure about this one but I went ahead and added it so we could discuss any implications I might have missed

Comment (tas50) [reply lib/chef/knife/bootstrap.rb]: we might just rename those templates altogether.

**Pull request merged**